src/KafkaNET.Library/Consumers/ConsumerIterator.cs

```diff
@@ -247,7 +247,7 @@ private TData MakeNext()
                 currentTopicInfo = currentDataChunk.TopicInfo;
                 Logger.DebugFormat("CurrentTopicInfo: ConsumedOffset({0}), FetchOffset({1})",
                                     currentTopicInfo.ConsumeOffset, currentTopicInfo.FetchOffset);
-                if (currentTopicInfo.ConsumeOffset != currentDataChunk.FetchOffset)
+                if (currentTopicInfo.FetchOffset < currentDataChunk.FetchOffset)
```

Comment:
My concern here is that you're now checking the currentTopicInfo.FetchOffset, but line 256 will update the currentTopicInfo.ConsumeOffset.  Perhaps this is ok, but I'm afraid this may introduce subtle bugs of the sort I found back in October between fetched & consumed offsets, only this time the consumer may miss messages if the ConsumeOffset gets set forward to the currentDataChunk.FetchOffset.